python/tests/test_tools/test_toolset.py

```diff
@@ -152,3 +154,89 @@ def test_api_key_missing() -> None:
         ),
     ):
         _ = toolset.workspace
+
+
+def test_processors(monkeypatch: pytest.MonkeyPatch) -> None:
+    """Test the `processors` field in `ComposioToolSet` constructor."""
+    preprocessor_called = postprocessor_called = False
+
+    def preprocess(request: dict) -> dict:
+        nonlocal preprocessor_called
+        preprocessor_called = True
+        return request
+
+    def postprocess(response: dict) -> dict:
+        nonlocal postprocessor_called
+        postprocessor_called = True
+        return response
+
+    with pytest.warns(DeprecationWarning):
+        toolset = ComposioToolSet(
+            processors={
+                "pre": {App.GMAIL: preprocess},
+                "post": {App.GMAIL: postprocess},
+            }
+        )
+    monkeypatch.setattr(toolset, "_execute_remote", lambda **_: {})
+
+    # Happy case
+    toolset.execute_action(action=Action.GMAIL_FETCH_EMAILS, params={})
+    assert preprocessor_called
+    assert postprocessor_called
+
+    # Improperly defined processors
+    preprocessor_called = postprocessor_called = False
+
+    def weird_postprocessor(reponse: dict) -> None:
```

Comment:
The variable name `reponse` in the `weird_postprocessor` function is misspelled. It should be `response`.